pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -435,8 +435,12 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		reconciler.EmitEvent(c.Recorder, before, after, pr)
 		return err
 	}
-
-	candidateTasks, err := dag.GetSchedulable(d, pipelineState.SuccessfulPipelineTaskNames()...)
+	// Get the next tasks to be executed which doesnt't have any TaskRuns created yet
+	// ExecutedPipelineTaskNames gets the list of finished tasks (either successful or failed or skipped)
```

Comment:
IIUC the skipped ones does not include those skipped because of a condition?
Having a runOn(["skipped"])  could be a trick to provide a way to express an `else` statement for a condition, which is something that we miss today, we need to define to opposite conditions for that.

Reply:
Yup, I was thinking that could be a solution to #1023

Reply:
yup, `runOn: [skip]` is equivalent to run this task if parent was not executed because of a condition failure i.e.

```
if condition then
    task1
else
    task2
```

Reply:
On second thoughts, runOn: [skip] might be not a bit loose, at least for handling conditionals. If we had this setup (sorry about the approximated YAML syntax):
```
task1:
  conditions:
    - condA
taskElseCondA:
    runOn: [ task1, skip ]

task2:
  conditions:
    - condB
taskElseCondB:
    runOn: [ task2, skip ]

```

If `condA` fails, task1 willl be skipped.  As a consequence `task2` fill be skipped too.
`taskElseCondA` will be executed, but what happens with `taskElseCondB`? 
Since `task2` was skipped, it could be executed. However the intention of the pipeline author there is probaly that `taskElseCondB` is executed if `condB` is false, which we haven't evaluated. 

I think it would be best to have the `else` task bound to the conditional rather than the task associated to the conditional.

Reply:
You are right task 2 will be skipped. One way to do what you want would be for task2 to say `runAfter: [task1: [`success`, `skip`]